eng/packages/http-client-csharp/generator/Azure.Generator/src/NamespaceVisitor.cs

```diff
@@ -57,7 +57,7 @@ private static void UpdateModelsNamespace(TypeProvider type)
             // TODO: need to take consideration of model-namespace configuration
             // if model-namespace is false, set namespace to $"{AzureClientPlugin.Instance.TypeFactory.RootNamespace}"
             // if model-namespace is true, set namespace to $"{AzureClientPlugin.Instance.TypeFactory.RootNamespace}.Models"
-            type.Type.Namespace = AzureClientPlugin.Instance.TypeFactory.GetCleanNameSpace($"{AzureClientPlugin.Instance.TypeFactory.RootNamespace}.Models");
+            type.Type.Namespace = AzureClientPlugin.Instance.TypeFactory.GetCleanNameSpace($"{AzureClientPlugin.Instance.TypeFactory.PackageName}.Models");
```

Comment:
I don't think we need to call GetCleanNamespace - we can possibly make this internal.

Reply:
This is mainly about the `Models` namespace, there is a model called `Models`, we update the namespace to `_Models`.
Will handle it properly in https://github.com/Azure/azure-sdk-for-net/issues/48275

Reply:
Ok, but we shouldn't need it when we are just directly using the PackageName.

Reply:
The problem is we are using `$"{AzureClientPlugin.Instance.TypeFactory.RootNamespace}.Models"`, we use GetCleanNameSpace to update it to `XXX._Models` when confliction happens.

Reply:
Yes, I understand that it is needed here. I was referring to the other comment where we are using RootNamespace alone.